source/types.ts

```diff
@@ -229,3 +225,22 @@ export interface QueryOptions {
 }
 
 export interface CallOptions extends MutationOptions, QueryOptions {}
+
+type IsEmptyType<T> = keyof T extends never ? true : false;
+type ExcludeNumber<T> = T extends number ? never : T;
+
+export interface ExtendibleEntityTypeMap {}
+interface DefaultEntityTypeMap {
+  [key: string]: {
+    [key: string]: any;
+  };
+}
+export type EntityTypeMap = IsEmptyType<ExtendibleEntityTypeMap> extends true
+  ? DefaultEntityTypeMap
+  : ExtendibleEntityTypeMap;
+
+export type EntityType = ExcludeNumber<keyof EntityTypeMap>;
+export type EntityData<TEntityType extends EntityType = EntityType> =
+  EntityTypeMap[TEntityType];
+
+export interface TypedContextSubtypeMap {}
```

Comment:
are we using all of these types still? don't quite understand their usages in that case, and could need some comments